pkg/bytecode/compiler.go

```diff
@@ -107,6 +111,14 @@ func (c *Compiler) compileBinaryExpression(expr *parser.BinaryExpression) error
 	if err := c.Compile(expr.Right); err != nil {
 		return err
 	}
+	if expr.Left.Type() == parser.NUM_TYPE && expr.Right.Type() == parser.NUM_TYPE {
+		return c.compileNumBinaryExpression(expr)
+	}
```

Comment:
This is not necessary for this PR, but I wanted to think ahead about how we would be outputting different opcodes based on type (e.g. OpStringConcat instead of OpAdd when two strings are involved)